### PR TITLE
[Challenge] Eevee only challenge

### PR DESCRIPTION
**Pull request by hayuna**
## What are the changes?
Add new challenge mode

## Why am I doing these changes?
Make challenge more interesting

## What did change?
Added new option in challenge
Allow to use only Eevee
Allow to use duplicated species
Not allow to use other pokemon than Eeveelutions

### Screenshots/Videos
New option in challenge menu
Screenshot 2024-07-02 at 10 50 44
Allow to use *only* Eevee
Screenshot 2024-07-02 at 10 50 59
Allow to have duplicated species
Screenshot 2024-07-02 at 10 51 14
Cannot use other pokemon than Eevee
Screenshot 2024-07-02 at 10 51 47
Completed challenge
Screenshot 2024-07-04 at 04 36 51

## How to test the changes?
Choose new challenge

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by hayuna** [comment]
Seems like this test is flaky, sometimes is working, sometimes fail

src/test/abilities/costar.test.ts > Abilities - COSTAR > ability copies positive stat changes

File: src/locales/fr/achv.ts [hunk 1]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "Hey ! Listen !",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only challenge.",
```
Comment (Adri1) [inline src/locales/fr/achv.ts:269]:
```suggestion
    name: "RP Pania",
    description: "Terminer un challenge Évoli.",
```

File: src/locales/fr/challenges.ts [hunk 2]
```diff
@@ -23,4 +23,10 @@ export const challenges: TranslationEntries = {
     "desc_default": "Vous ne pouvez choisir que des Pokémon du type sélectionné."
     //type in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee in this challenge",
+    "value.0": "Off",
+    "value.1": "On",
```
Comment (Adri1) [inline src/locales/fr/challenges.ts:30]:
```suggestion
    "name": "Évoli uniquement",
    "desc": "Vous ne pouvez choisir que des Évoli comme starters.",
    "value.0": "Désactivé",
    "value.1": "Activé",
```

**Review by Adri1 — COMMENTED** [review]
Added French translations

File: src/locales/ko/challenges.ts [hunk 3]
```diff
@@ -23,4 +23,10 @@ export const challenges: TranslationEntries = {
     "desc_default": "선택한 타입의 포켓몬만 사용할 수 있습니다."
     //type in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee in this challenge",
```
Comment (sodaMelon) [inline src/locales/ko/challenges.ts:28]:
```suggestion
    "name": "이브이 모드",
    "desc": "이 챌린지는 이브이만 사용할 수 있습니다.",
```

File: src/locales/ko/achv.ts [hunk 4]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "설마 자시안으로?",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only Eevee challenge.",
```
Comment (sodaMelon) [inline src/locales/ko/achv.ts:269]:
```suggestion
    name: "이브이 트레이너",
    description: "이브이 챌린지 모드 클리어.",
```

**Review by sodaMelon — COMMENTED** [review]
Really cool  the "Eevee only challenge"
i added korean localization.

File: src/locales/zh_CN/challenges.ts [hunk 5]
```diff
@@ -22,4 +22,10 @@ export const challenges: TranslationEntries = {
     "desc": "你只能使用{{type}}\n属性的宝可梦",
     "desc_default": "你只能使用所选\n属性的宝可梦"
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee in this challenge",
+    "value.0": "Off",
+    "value.1": "On",
```
Comment (RimKnight) [inline src/locales/zh_CN/challenges.ts:29]:
```suggestion
    "name": "伊布限定",
    "desc": "在这个挑战中你只能使用伊布",
    "value.0": "禁用",
    "value.1": "启用",
```

File: src/locales/de/achv.ts [hunk 6]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "Ein ewiges Abenteuer!",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only challenge.",
```
Comment (CodeTappert) [inline src/locales/de/achv.ts:269]:
```suggestion
    name: "E wie Evoli",
    description: "Beende die Nur-Evoli Herausforderung.",
```

File: src/locales/zh_CN/achv.ts [hunk 7]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "林克，醒醒！",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only challenge.",
```
Comment (RimKnight) [inline src/locales/zh_CN/achv.ts:269]:
```suggestion
    name: "唯布独尊",
    description: "完成伊布限定挑战。",
```

File: src/locales/de/challenges.ts [hunk 8]
```diff
@@ -23,4 +23,10 @@ export const challenges: TranslationEntries = {
     "desc_default": "Du kannst nur Pokémon des gewählten Typs verwenden."
     // types in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee in this challenge",
+    "value.0": "Off",
+    "value.1": "On",
+  }
```
Comment (CodeTappert) [inline src/locales/de/challenges.ts:31]:
```suggestion
  "eeveeOnly": {
    "name": "Nur-Evoli",
    "desc": "Du kannst nur Evoli in dieser Herausforderung verwenden.",
    "value.0": "Aus",
    "value.1": "An",
  }
```

**Comment by CodeTappert** [comment]
If Eveelutions are allowed then the description should reflect this.

**Review by RimKnight — COMMENTED** [review]
Add cn translation

**Comment by hayuna** [comment]
Moved On/Off to the top of translations
Changed description to be more descriptive with evolutions of Eevee

File: src/locales/pt_BR/achv.ts [hunk 9]
```diff
@@ -532,4 +532,8 @@ export const PGFachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "Clube das Winx",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only challenge.",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/achv.ts:537]:
```suggestion
    name: "Treinador de Eevee",
    description: "Complete o desafio Somente Eevees.",
```

File: src/locales/fr/challenges.ts [hunk 10]
```diff
@@ -23,4 +25,8 @@ export const challenges: TranslationEntries = {
     "desc_default": "Vous ne pouvez choisir que des Pokémon du type sélectionné."
     //type in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "Évoli uniquement",
+    "desc": "Vous ne pouvez choisir que des Évoli comme starters.",
```
Comment (Adri1) [inline src/locales/fr/challenges.ts:30]:
```suggestion
    "desc": "Vous ne pouvez utiliser que des Évoli et leurs évolutions.",
```

File: src/locales/pt_BR/challenges.ts [hunk 11]
```diff
@@ -1,6 +1,8 @@
 import { TranslationEntries } from "#app/interfaces/locales";
 
 export const challenges: TranslationEntries = {
+  "on": "On",
+  "off": "Off",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/challenges.ts:5]:
```suggestion
  "on": "Ligado",
  "off": "Desligado",
```

File: src/locales/ko/challenges.ts [hunk 12]
```diff
@@ -1,6 +1,8 @@
 import { TranslationEntries } from "#app/interfaces/locales";
 
 export const challenges: TranslationEntries = {
+  "on": "On",
+  "off": "Off",
```
Comment (sodaMelon) [inline src/locales/ko/challenges.ts:5]:
```suggestion
  "on": "설정",
  "off": "해제",
```

**Review by Adri1 — COMMENTED** [review]
Edited French (you can merge this edit direcly from this comment with the button on the bottom right of it, no copy-pasting needed)

File: src/locales/pt_BR/challenges.ts [hunk 13]
```diff
@@ -22,4 +24,8 @@ export const challenges: TranslationEntries = {
     "desc": "Você só pode user Pokémon do tipo {{type}}.",
     "desc_default": "Você só pode user Pokémon de um único tipo."
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee and its evolutions in this challenge",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/challenges.ts:29]:
```suggestion
    "desc": "Você só pode usar Pokémon do tipo {{type}}.",
    "desc_default": "Você só pode usar Pokémon de um único tipo."
  },
  "eeveeOnly": {
    "name": "Somente Eevees",
    "desc": "Você só pode usar Eevee e suas evoluções nesse desafio",
```

**Comment by hayuna** [comment]
Connected with https://github.com/pagefaultgames/pokerogue/issues/2708

**Review by returntoice — APPROVED** [review]

**Review by sodaMelon — APPROVED** [review]

**Comment by jaimefd** [comment]
I think this challenge should be extended to Monomon challenge, where you gotta bring 1-6 of the same species (depending on what weights allow ofc). Could reward a ribbon for the selected species if beaten.

**Comment by hayuna** [comment]
Eevee is a unique pokemon with many evolution lines, not sure if any other pokemon would be great for similar challenge

File: src/locales/ko/achv.ts [hunk 14]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "설마 자시안으로?",
   },
+  "EEVEE_TRAINER": {
+    name: "이브이 트레이너",
+    description: "이브이 챌린지 모드 클리어.",
```
Comment (EnochG1) [inline src/locales/ko/achv.ts:269]:
```suggestion
    description: "이브이 모드 챌린지 클리어.",
```
@sodaMelon 
챌린지의 이름이 '이브이 모드'라 어순을 변경했습니다.

File: src/locales/ko/challenges.ts [hunk 15]
```diff
@@ -23,4 +25,8 @@ export const challenges: TranslationEntries = {
     "desc_default": "선택한 타입의 포켓몬만 사용할 수 있습니다."
     //type in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "이브이 모드",
+    "desc": "이 챌린지는 이브이만 사용할 수 있습니다.",
```
Comment (EnochG1) [inline src/locales/ko/challenges.ts:30]:
```suggestion
    "desc": "이 챌린지는 이브이와 이브이의 진화체만 사용할 수 있습니다.",
```
changed korean translation for detail (CC : @sodaMelon)

**Review by RimKnight — APPROVED** [review]

File: src/locales/de/challenges.ts [hunk 16]
```diff
@@ -23,4 +25,8 @@ export const challenges: TranslationEntries = {
     "desc_default": "Du kannst nur Pokémon des gewählten Typs verwenden."
     // types in pokemon-info
   },
+  "eeveeOnly": {
+    "name": "Nur-Evoli",
+    "desc": "Du kannst nur Evoli in dieser Herausforderung verwenden.",
```
Comment (CodeTappert) [inline src/locales/de/challenges.ts:30]:
```suggestion
    "desc": "Du kannst nur Evoli und dessen Entwicklungen in dieser Herausforderung verwenden.",
```

File: src/locales/es/challenges.ts [hunk 17]
```diff
@@ -1,6 +1,8 @@
 import { TranslationEntries } from "#app/interfaces/locales";
 
 export const challenges: TranslationEntries = {
+  "on": "On",
+  "off": "Off",
```
Comment (GoldTra) [inline src/locales/es/challenges.ts:5]:
```suggestion
  "on": "Activado",
  "off": "Desactivado",
```

File: src/locales/es/achv.ts [hunk 18]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "Mono FAIRY",
   },
+  "EEVEE_TRAINER": {
+    name: "Eevee Trainer",
+    description: "Complete Eevee Only challenge.",
```
Comment (GoldTra) [inline src/locales/es/achv.ts:269]:
```suggestion
    name: "Entranador de Eevee",
    description: "Completa el desafío de solo Eevee.",
```

**Review by GoldTra — CHANGES_REQUESTED** [review]
Spanish translations

File: src/locales/es/achv.ts [hunk 19]
```diff
@@ -264,6 +264,10 @@ export const PGMachv: AchievementTranslationEntries = {
   "MONO_FAIRY": {
     name: "Mono FAIRY",
   },
+  "EEVEE_TRAINER": {
+    name: "Entranador de Eevee",
```
Comment (GoldTra) [inline src/locales/es/achv.ts:268]:
```suggestion
    name: "Entrenador de Eevee",
```

**Review by GoldTra — CHANGES_REQUESTED** [review]
Fix typo Spanish

**Comment by nayfaan** [comment]
In the Challenge modifier menu, instead of 

Eevee Only...............On / Off

maybe change it to something like

Others......................None / Eevee Only

?

This could allow future special challenges to be added to the "Others" menu without having them to all make a new row each.

**Comment by nayfaan** [comment]
And what happens if I selects "On" with Eevee only while the other two options are not at default value? e.g. Gen IX + mono Ghost selected

**Comment by hayuna** [comment]
@nayfaan I have compared current example.

Dark type + gen I = no pokemon available, so you cannot start

File: src/locales/es/challenges.ts [hunk 20]
```diff
@@ -22,4 +24,8 @@ export const challenges: TranslationEntries = {
     "desc": "Solo puedes usar Pokémon with the {{type}} type.",
     "desc_default": "Solo puedes usar Pokémon del tipo elegido.",
   },
+  "eeveeOnly": {
+    "name": "Eevee only",
+    "desc": "You can use only Eevee and its evolutions in this challenge",
+  }
```
Comment (InnocentGameDev) [inline src/locales/es/challenges.ts:30]:
```suggestion
  "eeveeOnly": {
    "name": "Solamente Eevee",
    "desc": "Solo puedes usar a Eevee o a sus evoluciones en este desafío",
  }
```

**Review by InnocentGameDev — CHANGES_REQUESTED** [review]
Some lines weren't translated

**Comment by NicusPulcis** [comment]
hey @hayuna i've added translations for both achv and challenges in italian!

**Comment by Naothe** [comment]
I'd like the third menu to be Others: Off / Eevee
This way there could be other challenges, for example: Fossil only

**Comment by CodeTappert** [comment]
Please fix conflicts

**Comment by hayuna** [comment]
New PR

https://github.com/pagefaultgames/pokerogue/pull/3305

**Pull request closed without merging**